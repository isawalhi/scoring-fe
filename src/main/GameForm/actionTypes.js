import { APP_NAME } from '../../constants';

export const DO_SUBMIT_FORM = `${APP_NAME}/DO_SUBMIT_FORM`;
