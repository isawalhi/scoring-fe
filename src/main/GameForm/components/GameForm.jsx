import React from 'react';
import { Form, Input, Button } from 'antd';

const GameForm = ({ onSubmit }) => {
  const onFinish = (values) => {
    onSubmit(values);
  };

  const onFinishFailed = (errorInfo) => {
    // eslint-disable-next-line no-console
    console.error('Failed:', errorInfo);
  };

  return (
    <Form onFinish={onFinish} onFinishFailed={onFinishFailed}>
      <Form.Item
        label="1st Place"
        name="firstPlace"
        rules={[
          {
            required: true,
            message: 'Please input first place player!',
          },
        ]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        label="2nd Place"
        name="secondPlace"
        rules={[
          {
            required: true,
            message: 'Please input first place player!',
          },
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item label="3rd Place" name="thirdPlace">
        <Input />
      </Form.Item>
      <Form.Item label="4th Place" name="forthPlace">
        <Input />
      </Form.Item>
      <Form.Item>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default GameForm;
