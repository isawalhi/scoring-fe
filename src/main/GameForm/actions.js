import * as Types from './actionTypes';

export const doSubmit = (values) => ({
  type: Types.DO_SUBMIT_FORM,
  values,
});
