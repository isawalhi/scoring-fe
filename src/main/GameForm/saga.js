import { all, call, takeLatest } from 'redux-saga/effects';
import lodashValues from 'lodash/values';
import compact from 'lodash/compact';
import * as Types from './actionTypes';
import { doPostRequest } from '../../HTTP/saga';
import { MICROSERVICES_DOMAINS } from '../constants';

const { GAMES_SERVICE } = MICROSERVICES_DOMAINS;

export function* submitGame(action) {
  const { values } = action;
  const payload = {
    standings: compact(lodashValues(values)),
  };
  try {
    yield call(
      doPostRequest,
      `${GAMES_SERVICE.DEFAULT_LOCAL_URL}/matches`,
      payload,
      {},
      {},
      true,
      ''
    ) || {};
  } catch (e) {
    // eslint-disable-next-line no-console
    console.error(e);
  }
}

export default function* watcher() {
  yield all([takeLatest(Types.DO_SUBMIT_FORM, submitGame)]);
}
