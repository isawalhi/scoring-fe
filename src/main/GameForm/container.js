import { connect } from 'react-redux';
import * as actions from './actions';
import GameForm from './components/GameForm';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
  onSubmit: (values) => dispatch(actions.doSubmit(values)),
});

export default connect(mapStateToProps, mapDispatchToProps)(GameForm);
