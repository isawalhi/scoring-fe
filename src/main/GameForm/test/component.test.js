import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GameForm from '../components/GameForm';

it('renders', () => {
  const wrapper = shallow(<GameForm />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
