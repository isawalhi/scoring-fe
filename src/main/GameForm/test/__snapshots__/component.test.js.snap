// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ForwardRef(InternalForm)
  onFinish={[Function]}
  onFinishFailed={[Function]}
>
  <FormItem
    label="1st Place"
    name="firstPlace"
    rules={
      Array [
        Object {
          "message": "Please input first place player!",
          "required": true,
        },
      ]
    }
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem
    label="2nd Place"
    name="secondPlace"
    rules={
      Array [
        Object {
          "message": "Please input first place player!",
          "required": true,
        },
      ]
    }
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem
    label="3rd Place"
    name="thirdPlace"
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem
    label="4th Place"
    name="forthPlace"
  >
    <Input
      type="text"
    />
  </FormItem>
  <FormItem>
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;
